WildFly - Attempt to register the same `ManagedExecutorDefinition` fails

This doesn't quite work, at least in WildFly. There are two tests, `ManagedExecutorDefinitionWebTests` and `ManagedScheduledExecutorDefinitionWebTests`, which attempt to register the same `ManagedExecutorDefinition` twice. It really seems they should be creating different resource names or the packages themselves cannot be added.

I will note it's not enough to just remove the package import as both the servlet and EJB context URL's are injected into the test via Arquillian.

One minor nit, the formatting is using a combo of tabs and spaces. It seems like ideally, it would be consistent.

_Originally posted by @jamezp in https://github.com/jakartaee/concurrency/pull/260#pullrequestreview-1055907581_

I would like this to be a TCK challenge. Could someone label it as such? I don't see how it could be valid to register 2 resources with the same JNDI name. In WildFly we see a stack trace like:
```
07:37:56,266 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-3) MSC000001: Failed to start service jboss.deployment.unit."ManagedScheduledExecutorDefinitionTests_web.war".INSTALL: org.jboss.msc.service.StartException in service jboss.deployment.unit."ManagedScheduledExecutorDefinitionTests_web.war".INSTALL: WFLYSRV0153: Failed to process phase INSTALL of deployment "ManagedScheduledExecutorDefinitionTests_web.war"
	at org.jboss.as.server@19.0.0.Beta15//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:189)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1363)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.jboss.as.server.deployment.DeploymentUnitProcessingException: org.jboss.msc.service.DuplicateServiceException: Service org.wildfly.ee.concurrent.scheduled-executor.ManagedScheduledExecutorDefinitionTests_web_ManagedScheduledExecutorDefinitionTests_web_ManagedScheduledExecutorDefinitionTests_web_java:module/concurrent/ScheduledExecutorB is already registered
	at org.jboss.as.ee@27.0.0.Alpha5-SNAPSHOT//org.jboss.as.ee.concurrent.resource.definition.ManagedScheduledExecutorDefinitionInjectionSource.getResourceValue(ManagedScheduledExecutorDefinitionInjectionSource.java:122)
	at org.jboss.as.ee@27.0.0.Alpha5-SNAPSHOT//org.jboss.as.ee.component.deployers.ModuleJndiBindingProcessor.addJndiBinding(ModuleJndiBindingProcessor.java:271)
	at org.jboss.as.ee@27.0.0.Alpha5-SNAPSHOT//org.jboss.as.ee.component.deployers.ModuleJndiBindingProcessor$1.handle(ModuleJndiBindingProcessor.java:242)
	at org.jboss.as.ee@27.0.0.Alpha5-SNAPSHOT//org.jboss.as.ee.component.ClassDescriptionTraversal.run(ClassDescriptionTraversal.java:54)
	at org.jboss.as.ee@27.0.0.Alpha5-SNAPSHOT//org.jboss.as.ee.component.deployers.ModuleJndiBindingProcessor.processClassConfigurations(ModuleJndiBindingProcessor.java:246)
	at org.jboss.as.ee@27.0.0.Alpha5-SNAPSHOT//org.jboss.as.ee.component.deployers.ModuleJndiBindingProcessor.deploy(ModuleJndiBindingProcessor.java:160)
	at org.jboss.as.server@19.0.0.Beta15//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:182)
	... 8 more
Caused by: org.jboss.msc.service.DuplicateServiceException: Service org.wildfly.ee.concurrent.scheduled-executor.ManagedScheduledExecutorDefinitionTests_web_ManagedScheduledExecutorDefinitionTests_web_ManagedScheduledExecutorDefinitionTests_web_java:module/concurrent/ScheduledExecutorB is already registered
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceRegistrationImpl.set(ServiceRegistrationImpl.java:135)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl.startInstallation(ServiceControllerImpl.java:243)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceContainerImpl.install(ServiceContainerImpl.java:758)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceTargetImpl.install(ServiceTargetImpl.java:260)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$ChildServiceTarget.install(ServiceControllerImpl.java:2065)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceTargetImpl.install(ServiceTargetImpl.java:260)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$ChildServiceTarget.install(ServiceControllerImpl.java:2065)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceBuilderImpl.install(ServiceBuilderImpl.java:192)
	at org.jboss.as.ee@27.0.0.Alpha5-SNAPSHOT//org.jboss.as.ee.concurrent.resource.definition.ManagedScheduledExecutorDefinitionInjectionSource.getResourceValue(ManagedScheduledExecutorDefinitionInjectionSource.java:101)
	... 14 more
```
The cause being the duplicate resource name being registered.

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/resourcedef/ManagedExecutorDefinitionTests.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/resourcedef/ManagedExecutorDefinitionTests.java
@@ -64,6 +64,7 @@ public class ManagedExecutorDefinitionTests extends TestClient{
 				.addPackages(false, getFrameworkPackage(), ManagedExecutorDefinitionTests.class.getPackage())
 				.deleteClasses(
 						AppBean.class,
+						ManagedExecutorDefinitionWebBean.class,
 						ManagedExecutorDefinitionServlet.class,
 						ManagedExecutorDefinitionOnEJBServlet.class,
 						ContextServiceDefinitionServlet.class)

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/resourcedef/ManagedExecutorDefinitionWebBean.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/resourcedef/ManagedExecutorDefinitionWebBean.java
@@ -1,0 +1,42 @@
+/*
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package ee.jakarta.tck.concurrent.spec.ManagedExecutorService.resourcedef;
+
+import jakarta.ejb.Local;
+import jakarta.ejb.Stateless;
+import jakarta.enterprise.concurrent.ManagedExecutorDefinition;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+/**
+ * ContextServiceDefinitions are defined under
+ * {@link ee.jakarta.tck.concurrent.spec.ContextService.contextPropagate.ContextServiceDefinitionBean}
+ */
+@ManagedExecutorDefinition(name = "java:app/concurrent/EJBExecutorA",
+                           context = "java:app/concurrent/EJBContextA",
+                           maxAsync = 2,
+                           hungTaskThreshold = 300000)
+@ManagedExecutorDefinition(name = "java:comp/concurrent/EJBExecutorC")
+@Local(ManagedExecutorDefinitionInterface.class)
+@Stateless
+public class ManagedExecutorDefinitionWebBean implements ManagedExecutorDefinitionInterface {
+	
+	@Override
+	public Object doLookup(String name) throws NamingException {
+		return InitialContext.doLookup(name);
+	}
+}

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/resourcedef/ManagedExecutorDefinitionWebTests.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/resourcedef/ManagedExecutorDefinitionWebTests.java
@@ -58,6 +58,7 @@ public class ManagedExecutorDefinitionWebTests extends TestClient{
 						ContextServiceDefinitionInterface.class,
 						ContextServiceDefinitionWebBean.class,
 						ContextServiceDefinitionServlet.class)
+				.deleteClasses(ManagedExecutorDefinitionBean.class)
 				.addAsServiceProvider(ThreadContextProvider.class.getName(), IntContextProvider.class.getName(), StringContextProvider.class.getName());
 		
 		return war;

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/resourcedef/ManagedScheduledExecutorDefinitionTests.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/resourcedef/ManagedScheduledExecutorDefinitionTests.java
@@ -65,13 +65,13 @@ public class ManagedScheduledExecutorDefinitionTests extends TestClient {
 				.addPackages(false, getFrameworkPackage(), ManagedScheduledExecutorDefinitionTests.class.getPackage())
 				.deleteClasses(
 						ReqBean.class,
+						ManagedScheduledExecutorDefinitionWebBean.class,
 						ManagedScheduledExecutorDefinitionServlet.class,
 						ManagedScheduledExecutorDefinitionOnEJBServlet.class)
 				.addClasses(
 						ContextServiceDefinitionInterface.class,
 						ContextServiceDefinitionBean.class)
 				.addAsManifestResource(ManagedScheduledExecutorDefinitionTests.class.getPackage(), "ejb-jar.xml", "ejb-jar.xml");
-//				TODO document how users can dynamically inject vendor specific deployment descriptors into this archive
 		
 		EnterpriseArchive ear = ShrinkWrap.create(EnterpriseArchive.class, "ManagedScheduledExecutorDefinitionTests.ear").addAsModules(war, jar);
 		

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/resourcedef/ManagedScheduledExecutorDefinitionWebBean.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/resourcedef/ManagedScheduledExecutorDefinitionWebBean.java
@@ -1,0 +1,43 @@
+/*
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package ee.jakarta.tck.concurrent.spec.ManagedScheduledExecutorService.resourcedef;
+
+import jakarta.ejb.Local;
+import jakarta.ejb.Stateless;
+import jakarta.enterprise.concurrent.ManagedScheduledExecutorDefinition;
+import ee.jakarta.tck.concurrent.spec.ContextService.contextPropagate.ContextServiceDefinitionServlet;
+import ee.jakarta.tck.concurrent.spec.ContextService.contextPropagate.ContextServiceDefinitionBean;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+/**
+ * @ContextServiceDefinitions are defined under {@link ContextServiceDefinitionServlet} and {@link ContextServiceDefinitionBean}
+ */
+@ManagedScheduledExecutorDefinition(name = "java:app/concurrent/EJBScheduledExecutorA",
+                           context = "java:app/concurrent/EJBContextA",
+                           maxAsync = 3,
+                           hungTaskThreshold = 360000)
+@ManagedScheduledExecutorDefinition(name = "java:comp/concurrent/EJBScheduledExecutorC")
+@Local(ManagedScheduleExecutorDefinitionInterface.class)
+@Stateless
+public class ManagedScheduledExecutorDefinitionWebBean implements ManagedScheduleExecutorDefinitionInterface {
+	
+	@Override
+	public Object doLookup(String name) throws NamingException {
+		return InitialContext.doLookup(name);
+	}
+}

--- a/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/resourcedef/ManagedScheduledExecutorDefinitionWebTests.java
+++ b/tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedScheduledExecutorService/resourcedef/ManagedScheduledExecutorDefinitionWebTests.java
@@ -57,7 +57,7 @@ public class ManagedScheduledExecutorDefinitionWebTests extends TestClient {
 						ContextServiceDefinitionServlet.class,
 						ContextServiceDefinitionInterface.class,
 						ContextServiceDefinitionWebBean.class)
-				.addAsWebInfResource(ManagedScheduledExecutorDefinitionWebTests.class.getPackage(), "ejb-jar.xml", "ejb-jar.xml")
+				.deleteClasses(ManagedScheduledExecutorDefinitionBean.class)
 				.addAsServiceProvider(ThreadContextProvider.class.getName(), IntContextProvider.class.getName(), StringContextProvider.class.getName());
 		
 		return war;

